main/src/main/scala/sbt/ScriptedRun.scala

```diff
@@ -0,0 +1,179 @@
+/*
+ * sbt
+ * Copyright 2011 - 2018, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * Licensed under Apache License 2.0 (see LICENSE)
+ */
+
+package sbt
+
+import java.io.File
+import java.lang.reflect.Method
+import scala.annotation.unused
+
+sealed trait ScriptedRun {
```

Comment:
Maybe it would be good to hide this under `package[sbt]` and `package sbt.internal`?